react-ui/src/pages/submission/SubmissionForm.jsx

```diff
@@ -357,33 +356,45 @@ function SubmissionForm({
     }
   };
 
-  const disableSubmitting = submitting || isMutating;
+  const nameCharRemaining = CHAR_LIMIT_NAME - formState.name.value.length;
+  const descriptionCharRemaining = CHAR_LIMIT_DESCRIPTION - formState.description.value.length;
+  const disableEditing = submitting || isMutating;
+  const disableSubmitting = disableEditing || nameCharRemaining < 0 || descriptionCharRemaining < 0;
 
   const classes = useStyles();
 
   return (
-    <ProtectedRoute
-      message="You must log in with Reddit to submit a flag"
-      showCancel={false}
-    >
+    <ProtectedRoute message="You must log in with Reddit to submit a flag" showCancel={false}>
       <form id="submission-form">
-        <Fieldset disabled={disableSubmitting || submissionExpired}>
-          <TextField
-            id="username"
-            variant="filled"
-            label="Username"
-            disabled
-            value={username}
-          />
+        <Fieldset disabled={disableEditing || submissionExpired}>
+          <TextField id="username" variant="filled" label="Username" disabled value={username} />
           <TextField
             id="name"
             name="name"
             color="secondary"
             variant="filled"
-            helperText={formState.name.error || 'A concise name for your flag'}
+            helperText={(
+              <div>
+                <Typography display="inline">
+                  {formState.name.error || 'A concise name for your flag'}
+                  .
+                  {' '}
+                </Typography>
+                <Typography display="inline">
+                  Character limit:
+                  {' '}
+                  {CHAR_LIMIT_NAME}
+                  {' '}
+                  (remaining:
+                  {' '}
+                  {nameCharRemaining}
+                  )
```

Comment:
Maybe just a `charCount / CHAR_LIMIT` would be sufficient here. I think character counts on input fields are a common enough pattern that we don't need the text. This is what my Reddit comment input looks like: 
![image](https://github.com/heshammourad/vexillology-contests/assets/21203724/a63b6f8a-7a68-41a3-aedc-f1abcbef7e82)

I know that's different than what I said about the flag counter, but I feel like that may not be as intuitive if you don't interact with it, while this will change noticeably as you type.